off-chain/test/Examples/Addition/ContractSeath.purs

```diff
@@ -1,101 +1,93 @@
-mainTest config = runPlutipContract config distribution $
-  \((admin /\ leader') /\ participants') -> do
-    -- contract initialization by some admin
-    _ <- withKeyWallet admin initialSeathContract
+mainTest :: RunnerConfig AdditionDatum -> Contract Unit
+mainTest config = do
+  -- todo: check that parties participants have enough funds by config.minAdaRequired
+  logInfo' $ "Running with " <> runnerConfInfo config
+
+  let
+    leaderKeyWallet = (unwrap config).seathLeader
+    leader = Leader leaderKeyWallet
 
+  unless ((unwrap config).alreadyInitialized) $ do
+    -- contract initialization by some admin
+    _ <- withKeyWallet leaderKeyWallet initialSeathContract
     logInfo' "----------------------- INIT DONE -------------------------"
 
-    -- Seath round logic
-    vaildatorHash <- fixedValidatorHash
-    let leader = Leader leader'
-    leaderPublicKeyHash <- getPublicKeyHash leader'
-    logInfo' $ "@@ participants: " <> show (length participants')
-    let
-      participants = map Participant participants'
-      seathConfig = SeathConfig
-        { leader: leaderPublicKeyHash
-        , stateVaildatorHash: vaildatorHash
-        , actionHandler: handleAction
-        , queryBlockchainState: queryBlockchainState
-        }
-      logState = logBlockchainState leader participants vaildatorHash
-
-    logState -- log state after initialisation
-
-    actions <- SeathSetup.genUserActions participants
-    logInfo' $ "User actions: " <> show actions
-
-    (finalizedTxsAndActions /\ _) <- withKeyWallet leader' $ buildChain
-      seathConfig
-      actions
-      Nothing
-    let finalizedTxs /\ _ = unzip finalizedTxsAndActions
-    -- logInfo' $ "BuildChainResult: " <> show finalizedTxs
-    txIds <- SeathSetup.submitChain leader participants finalizedTxs logState
-
-    case last txIds of
-      Nothing -> throwContractError
-        "No IDs vere received after chain submission. Something is wrong."
-      Just txId -> do
-        awaitTxConfirmed txId
-        logState
-        checkFinalState leader participants vaildatorHash
-
-    logInfo' "end"
-  where
+  -- Seath round logic
+  vaildatorHash <- fixedValidatorHash
+
+  leaderPublicKeyHash <- getPublicKeyHash leaderKeyWallet
+  let
+    participants = NE.toArray $ map Participant
+      (unwrap config).seathParticipants
+    seathConfig = SeathConfig
+      { leader: leaderPublicKeyHash
+      , stateVaildatorHash: vaildatorHash
+      , actionHandler: handleAction
+      , queryBlockchainState: queryBlockchainState
+      }
+    logState = logBlockchainState leader participants vaildatorHash
+
+  logState -- log state after initialisation
+
+  actions <- SeathSetup.genUserActions participants
+  logInfo' $ "User actions: " <> show actions
+
+  (finalizedTxsAndActions /\ _) <- withKeyWallet leaderKeyWallet $ buildChain
+    seathConfig
+    actions
+    Nothing
+  let finalizedTxs /\ _ = unzip finalizedTxsAndActions
+  -- logInfo' $ "BuildChainResult: " <> show finalizedTxs
```

Comment:
For the demo, shall we use something like this but with much short output?

Reply:
Yeah, probably. Lets move discussion here #22